crawler/regelleistung.py

```diff
@@ -0,0 +1,297 @@
+    product_split_array = (df["product"].str.split('_')).to_numpy()
+    hours_from = np.array([product_list[1] for product_list in product_split_array])
+    hours_to = np.array([product_list[2] for product_list in product_split_array])
+    timedelta_from = np.array([timedelta(hours=int(hour)) for hour in hours_from])
+    timedelta_to = np.array([timedelta(hours=int(hour)) for hour in hours_to])
+    df['date_from'] = df["date_from"] + pd.to_timedelta(timedelta_from, 'd')
+    df['date_to'] = df["date_to"] + pd.to_timedelta(timedelta_to, 'd')
+
+    # adapt mw column to mwh column
+    hours_from = np.array([product_list[1] for product_list in product_split_array])
+    hours_to = np.array([product_list[2] for product_list in product_split_array])
+    hours_from_int = (hours_from).astype(np.int16)
+    hours_to_int = (hours_to).astype(np.int16)
+    hours_diff = hours_to_int - hours_from_int
+    cols_to_adapt = ['total_min_capacity_price_eur_mw', 
+                     'total_average_capacity_price_eur_mw',
+                     'total_marginal_capacity_price_eur_mw', 
+                     'germany_min_capacity_price_eur_mw', 
+                     'germany_average_capacity_price_eur_mw', 
+                     'germany_marginal_capacity_price_eur_mw', 
+                     'austria_min_capacity_price_eur_mw', 
+                     'austria_average_capacity_price_eur_mw', 
+                     'austria_marginal_capacity_price_eur_mw',
+                     'capacity_price_eur_mw']
+    for col_to_adapt in cols_to_adapt:
+        if col_to_adapt in df.columns:
+            final_col_name = col_to_adapt + 'h'
+            df[final_col_name] = pd.to_numeric(df[col_to_adapt]) / hours_diff
+
+    cols_to_drop = ['total_min_energy_price_eur_mwh', 
+                    'total_average_energy_price_eur_mwh', 
+                    'total_marginal_energy_price_eur_mwh',
+                    'germany_min_energy_price_eur_mwh',
+                    'germany_average_energy_price_eur_mwh',
+                    'germany_marginal_energy_price_eur_mwh', 
+                    'austria_min_energy_price_eur_mwh', 
+                    'austria_average_energy_price_eur_mwh',
+                    'austria_marginal_energy_price_eur_mwh',
+                    'total_min_capacity_price_eur_mw', 
+                    'total_average_capacity_price_eur_mw',
+                    'total_marginal_capacity_price_eur_mw', 
+                    'germany_min_capacity_price_eur_mw', 
+                    'germany_average_capacity_price_eur_mw', 
+                    'germany_marginal_capacity_price_eur_mw', 
+                    'austria_min_capacity_price_eur_mw', 
+                    'austria_average_capacity_price_eur_mw', 
+                    'austria_marginal_capacity_price_eur_mw',
+                    'capacity_price_eur_mw']
+    for col_to_drop in cols_to_drop:
+        if col_to_drop in df.columns:
+            df = df.drop(col_to_drop, axis=1)
```

Comment:
why was it necessary to drop these columns? just curious :)

Reply:
The first part of the columns were in all cases just n.a. and in parallel there existed the same columns with the name capacity instead of energy, which had filled values, so I deleted the others. In addition they also were deleted from regelleistung.net after the 2021-12-31. For the other columns I changed the unit from MW to MWh so I added a new column for them and deleted these ones.